raft/rafttest: TestRestart failed

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16125659?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16125659?buildTab=artifacts#/) on master @ [46854026bc11b3870c8bbefc976f5550b18dd82c](https://github.com/cockroachdb/cockroach/commits/46854026bc11b3870c8bbefc976f5550b18dd82c):


```
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/07/18 04:23:33 INFO: 5 is starting a new election at term 2
raft2024/07/18 04:23:33 INFO: 5 became candidate at term 3
raft2024/07/18 04:23:33 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 1 at term 3
raft2024/07/18 04:23:33 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 2 at term 3
raft2024/07/18 04:23:33 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 3 at term 3
raft2024/07/18 04:23:33 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 4 at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 5 lost leader 2 at term 3
raft2024/07/18 04:23:33 INFO: 5 received MsgVoteResp from 5 at term 3
raft2024/07/18 04:23:33 INFO: 5 has received 1 MsgVoteResp votes and 0 vote rejections
raft2024/07/18 04:23:33 INFO: 1 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/07/18 04:23:33 INFO: 1 became follower at term 3
raft2024/07/18 04:23:33 INFO: 1 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 1 lost leader 2 at term 3
raft2024/07/18 04:23:33 INFO: 4 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/07/18 04:23:33 INFO: 4 became follower at term 3
raft2024/07/18 04:23:33 INFO: 4 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 4 lost leader 2 at term 3
raft2024/07/18 04:23:33 INFO: 5 received MsgVoteResp from 1 at term 3
raft2024/07/18 04:23:33 INFO: 5 has received 2 MsgVoteResp votes and 0 vote rejections
raft2024/07/18 04:23:33 INFO: 3 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/07/18 04:23:33 INFO: 3 became follower at term 3
raft2024/07/18 04:23:33 INFO: 3 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 3 lost leader 2 at term 3
raft2024/07/18 04:23:33 INFO: 2 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/07/18 04:23:33 INFO: 2 became follower at term 3
raft2024/07/18 04:23:33 INFO: 2 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 2 lost leader 2 at term 3
raft2024/07/18 04:23:33 INFO: 5 received MsgVoteResp from 3 at term 3
raft2024/07/18 04:23:33 INFO: 5 has received 3 MsgVoteResp votes and 0 vote rejections
raft2024/07/18 04:23:33 INFO: 5 became leader at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 5 elected leader 5 at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 2 elected leader 5 at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 1 elected leader 5 at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 3 elected leader 5 at term 3
raft2024/07/18 04:23:33 INFO: raft.node: 4 elected leader 5 at term 3
    node_test.go:91: commits failed to converge!
I240718 04:23:38.520629 388 (gostd) node.go:100  [-] 8  raft.1: stop
I240718 04:23:38.520684 616 (gostd) node.go:100  [-] 9  raft.2: stop
I240718 04:23:38.520698 614 (gostd) node.go:100  [-] 10  raft.3: stop
I240718 04:23:38.520708 394 (gostd) node.go:100  [-] 11  raft.4: stop
I240718 04:23:38.521851 396 (gostd) node.go:100  [-] 12  raft.5: stop
--- FAIL: TestRestart (5.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40428

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16312986?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16312986?buildTab=artifacts#/) on master @ [4ae4a52772b4b6b2bf2c340e06919c090894dffb](https://github.com/cockroachdb/cockroach/commits/4ae4a52772b4b6b2bf2c340e06919c090894dffb):


```
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 3 not forwarding to itself at term 2; dropping proposal
raft2024/08/03 00:03:03 INFO: 4 is starting a new election at term 2
raft2024/08/03 00:03:03 INFO: 4 became candidate at term 3
raft2024/08/03 00:03:03 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 1 at term 3
raft2024/08/03 00:03:03 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 2 at term 3
raft2024/08/03 00:03:03 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 3 at term 3
raft2024/08/03 00:03:03 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 5 at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 4 lost leader 3 at term 3
raft2024/08/03 00:03:03 INFO: 4 received MsgVoteResp from 4 at term 3
raft2024/08/03 00:03:03 INFO: 4 has received 1 MsgVoteResp votes and 0 vote rejections
raft2024/08/03 00:03:03 INFO: 5 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/03 00:03:03 INFO: 5 became follower at term 3
raft2024/08/03 00:03:03 INFO: 5 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 5 lost leader 3 at term 3
raft2024/08/03 00:03:03 INFO: 3 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/03 00:03:03 INFO: 3 became follower at term 3
raft2024/08/03 00:03:03 INFO: 3 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 3 lost leader 3 at term 3
raft2024/08/03 00:03:03 INFO: 1 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/03 00:03:03 INFO: 1 became follower at term 3
raft2024/08/03 00:03:03 INFO: 1 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 1 lost leader 3 at term 3
raft2024/08/03 00:03:03 INFO: 2 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/03 00:03:03 INFO: 2 became follower at term 3
raft2024/08/03 00:03:03 INFO: 2 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 2 lost leader 3 at term 3
raft2024/08/03 00:03:03 INFO: 4 received MsgVoteResp from 3 at term 3
raft2024/08/03 00:03:03 INFO: 4 has received 2 MsgVoteResp votes and 0 vote rejections
raft2024/08/03 00:03:03 INFO: 4 received MsgVoteResp from 5 at term 3
raft2024/08/03 00:03:03 INFO: 4 has received 3 MsgVoteResp votes and 0 vote rejections
raft2024/08/03 00:03:03 INFO: 4 became leader at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 4 elected leader 4 at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 5 elected leader 4 at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 3 elected leader 4 at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 1 elected leader 4 at term 3
raft2024/08/03 00:03:03 INFO: raft.node: 2 elected leader 4 at term 3
    node_test.go:91: commits failed to converge!
I240803 00:03:08.972291 1723 (gostd) node.go:100  [-] 8  raft.1: stop
I240803 00:03:08.972370 1817 (gostd) node.go:100  [-] 9  raft.2: stop
I240803 00:03:08.972385 1815 (gostd) node.go:100  [-] 10  raft.3: stop
I240803 00:03:08.974438 1729 (gostd) node.go:100  [-] 11  raft.4: stop
I240803 00:03:08.975517 1731 (gostd) node.go:100  [-] 12  raft.5: stop
--- FAIL: TestRestart (5.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16383469?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16383469?buildTab=artifacts#/) on master @ [167a9af39d0c1fadb9de8ee3c40e616e6c98af78](https://github.com/cockroachdb/cockroach/commits/167a9af39d0c1fadb9de8ee3c40e616e6c98af78):


```
I240808 19:00:45.050083 1756 (gostd) node.go:100  [-] 9  raft.2: stop
I240808 19:00:45.050159 1754 (gostd) node.go:100  [-] 10  raft.3: stop
I240808 19:00:45.051214 1668 (gostd) node.go:100  [-] 11  raft.4: stop
I240808 19:00:45.051264 1670 (gostd) node.go:100  [-] 12  raft.5: stop
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=()
raft2024/08/08 19:00:45 INFO: 1 became follower at term 0
raft2024/08/08 19:00:45 INFO: newRaft 1 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/08 19:00:45 INFO: 1 became follower at term 1
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=(1)
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=(1 2)
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=(1 2 3)
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=(1 2 3 4)
raft2024/08/08 19:00:45 INFO: 1 switched to configuration voters=(1 2 3 4 5)
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=()
raft2024/08/08 19:00:45 INFO: 2 became follower at term 0
raft2024/08/08 19:00:45 INFO: newRaft 2 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/08 19:00:45 INFO: 2 became follower at term 1
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=(1)
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=(1 2)
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=(1 2 3)
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=(1 2 3 4)
raft2024/08/08 19:00:45 INFO: 2 switched to configuration voters=(1 2 3 4 5)
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=()
raft2024/08/08 19:00:45 INFO: 3 became follower at term 0
raft2024/08/08 19:00:45 INFO: newRaft 3 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/08 19:00:45 INFO: 3 became follower at term 1
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=(1)
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=(1 2)
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=(1 2 3)
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=(1 2 3 4)
raft2024/08/08 19:00:45 INFO: 3 switched to configuration voters=(1 2 3 4 5)
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=()
raft2024/08/08 19:00:45 INFO: 4 became follower at term 0
raft2024/08/08 19:00:45 INFO: newRaft 4 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/08 19:00:45 INFO: 4 became follower at term 1
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=(1)
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=(1 2)
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=(1 2 3)
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=(1 2 3 4)
raft2024/08/08 19:00:45 INFO: 4 switched to configuration voters=(1 2 3 4 5)
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=()
raft2024/08/08 19:00:45 INFO: 5 became follower at term 0
raft2024/08/08 19:00:45 INFO: newRaft 5 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/08 19:00:45 INFO: 5 became follower at term 1
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=(1)
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=(1 2)
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=(1 2 3)
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=(1 2 3 4)
raft2024/08/08 19:00:45 INFO: 5 switched to configuration voters=(1 2 3 4 5)
--- FAIL: TestRestart (5.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16451005?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16451005?buildTab=artifacts#/) on master @ [bb2613e5348ea864c6a9cc348af73117d701f453](https://github.com/cockroachdb/cockroach/commits/bb2613e5348ea864c6a9cc348af73117d701f453):


```
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/13 16:49:54 INFO: 4 is starting a new election at term 2
raft2024/08/13 16:49:54 INFO: 4 became candidate at term 3
raft2024/08/13 16:49:54 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 1 at term 3
raft2024/08/13 16:49:54 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 2 at term 3
raft2024/08/13 16:49:54 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 3 at term 3
raft2024/08/13 16:49:54 INFO: 4 [logterm: 2, index: 6] sent MsgVote request to 5 at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 4 lost leader 2 at term 3
raft2024/08/13 16:49:54 INFO: 4 received MsgVoteResp from 4 at term 3
raft2024/08/13 16:49:54 INFO: 4 has received 1 MsgVoteResp votes and 0 vote rejections
raft2024/08/13 16:49:54 INFO: 1 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/13 16:49:54 INFO: 1 became follower at term 3
raft2024/08/13 16:49:54 INFO: 1 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 1 lost leader 2 at term 3
raft2024/08/13 16:49:54 INFO: 5 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/13 16:49:54 INFO: 5 became follower at term 3
raft2024/08/13 16:49:54 INFO: 5 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 5 lost leader 2 at term 3
raft2024/08/13 16:49:54 INFO: 3 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/13 16:49:54 INFO: 3 became follower at term 3
raft2024/08/13 16:49:54 INFO: 3 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 3 lost leader 2 at term 3
raft2024/08/13 16:49:54 INFO: 4 received MsgVoteResp from 3 at term 3
raft2024/08/13 16:49:54 INFO: 4 has received 2 MsgVoteResp votes and 0 vote rejections
raft2024/08/13 16:49:54 INFO: 2 [term: 2] received a MsgVote message with higher term from 4 [term: 3]
raft2024/08/13 16:49:54 INFO: 2 became follower at term 3
raft2024/08/13 16:49:54 INFO: 2 [logterm: 2, index: 6, vote: 0] cast MsgVote for 4 [logterm: 2, index: 6] at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 2 lost leader 2 at term 3
raft2024/08/13 16:49:54 INFO: 4 received MsgVoteResp from 2 at term 3
raft2024/08/13 16:49:54 INFO: 4 has received 3 MsgVoteResp votes and 0 vote rejections
raft2024/08/13 16:49:54 INFO: 4 became leader at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 4 elected leader 4 at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 1 elected leader 4 at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 5 elected leader 4 at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 2 elected leader 4 at term 3
raft2024/08/13 16:49:54 INFO: raft.node: 3 elected leader 4 at term 3
I240813 16:49:59.527205 1604 (gostd) node.go:102  [-] 8  raft.1: stop
I240813 16:49:59.527268 716 (gostd) node.go:102  [-] 9  raft.2: stop
I240813 16:49:59.527281 714 (gostd) node.go:102  [-] 10  raft.3: stop
I240813 16:49:59.527339 1610 (gostd) node.go:102  [-] 11  raft.4: stop
    node_test.go:91: commits failed to converge!
--- FAIL: TestRestart (5.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16507366?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16507366?buildTab=artifacts#/) on master @ [1993fc04b5116f20f4814d637c7ce87b003687e4](https://github.com/cockroachdb/cockroach/commits/1993fc04b5116f20f4814d637c7ce87b003687e4):


```
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/17 04:09:52 INFO: 5 is starting a new election at term 2
raft2024/08/17 04:09:52 INFO: 5 became candidate at term 3
raft2024/08/17 04:09:52 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 1 at term 3
raft2024/08/17 04:09:52 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 2 at term 3
raft2024/08/17 04:09:52 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 3 at term 3
raft2024/08/17 04:09:52 INFO: 5 [logterm: 2, index: 6] sent MsgVote request to 4 at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 5 lost leader 2 at term 3
raft2024/08/17 04:09:52 INFO: 5 received MsgVoteResp from 5 at term 3
raft2024/08/17 04:09:52 INFO: 5 has received 1 MsgVoteResp votes and 0 vote rejections
raft2024/08/17 04:09:52 INFO: 2 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/08/17 04:09:52 INFO: 2 became follower at term 3
raft2024/08/17 04:09:52 INFO: 2 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 2 lost leader 2 at term 3
raft2024/08/17 04:09:52 INFO: 4 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/08/17 04:09:52 INFO: 4 became follower at term 3
raft2024/08/17 04:09:52 INFO: 4 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 4 lost leader 2 at term 3
raft2024/08/17 04:09:52 INFO: 1 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/08/17 04:09:52 INFO: 1 became follower at term 3
raft2024/08/17 04:09:52 INFO: 1 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 1 lost leader 2 at term 3
raft2024/08/17 04:09:52 INFO: 5 received MsgVoteResp from 2 at term 3
raft2024/08/17 04:09:52 INFO: 5 has received 2 MsgVoteResp votes and 0 vote rejections
raft2024/08/17 04:09:52 INFO: 3 [term: 2] received a MsgVote message with higher term from 5 [term: 3]
raft2024/08/17 04:09:52 INFO: 3 became follower at term 3
raft2024/08/17 04:09:52 INFO: 3 [logterm: 2, index: 6, vote: 0] cast MsgVote for 5 [logterm: 2, index: 6] at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 3 lost leader 2 at term 3
raft2024/08/17 04:09:52 INFO: 5 received MsgVoteResp from 1 at term 3
raft2024/08/17 04:09:52 INFO: 5 has received 3 MsgVoteResp votes and 0 vote rejections
raft2024/08/17 04:09:52 INFO: 5 became leader at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 5 elected leader 5 at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 1 elected leader 5 at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 4 elected leader 5 at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 2 elected leader 5 at term 3
raft2024/08/17 04:09:52 INFO: raft.node: 3 elected leader 5 at term 3
    node_test.go:91: commits failed to converge!
I240817 04:09:57.144339 948 (gostd) node.go:102  [-] 8  raft.1: stop
I240817 04:09:57.145540 1014 (gostd) node.go:102  [-] 9  raft.2: stop
I240817 04:09:57.145629 1012 (gostd) node.go:102  [-] 10  raft.3: stop
--- FAIL: TestRestart (5.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16641713?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16641713?buildTab=artifacts#/) on master @ [972ccd5d17c54f2cef84e38d14290e52b01bd70a](https://github.com/cockroachdb/cockroach/commits/972ccd5d17c54f2cef84e38d14290e52b01bd70a):


```
I240827 19:44:11.128486 2418 (gostd) node.go:102  [-] 10  raft.3: stop
I240827 19:44:11.128499 2353 (gostd) node.go:102  [-] 11  raft.4: stop
I240827 19:44:11.128510 2355 (gostd) node.go:102  [-] 12  raft.5: stop
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=()
raft2024/08/27 19:44:11 INFO: 1 became follower at term 0
raft2024/08/27 19:44:11 INFO: newRaft 1 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/27 19:44:11 INFO: 1 became follower at term 1
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=(1)
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=(1 2)
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=(1 2 3)
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=(1 2 3 4)
raft2024/08/27 19:44:11 INFO: 1 switched to configuration voters=(1 2 3 4 5)
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=()
raft2024/08/27 19:44:11 INFO: 2 became follower at term 0
raft2024/08/27 19:44:11 INFO: newRaft 2 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/27 19:44:11 INFO: 2 became follower at term 1
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=(1)
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=(1 2)
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=(1 2 3)
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=(1 2 3 4)
raft2024/08/27 19:44:11 INFO: 2 switched to configuration voters=(1 2 3 4 5)
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=()
raft2024/08/27 19:44:11 INFO: 3 became follower at term 0
raft2024/08/27 19:44:11 INFO: newRaft 3 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/27 19:44:11 INFO: 3 became follower at term 1
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=(1)
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=(1 2)
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=(1 2 3)
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=(1 2 3 4)
raft2024/08/27 19:44:11 INFO: 3 switched to configuration voters=(1 2 3 4 5)
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=()
raft2024/08/27 19:44:11 INFO: 4 became follower at term 0
raft2024/08/27 19:44:11 INFO: newRaft 4 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/27 19:44:11 INFO: 4 became follower at term 1
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=(1)
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=(1 2)
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=(1 2 3)
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=(1 2 3 4)
raft2024/08/27 19:44:11 INFO: 4 switched to configuration voters=(1 2 3 4 5)
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=()
raft2024/08/27 19:44:11 INFO: 5 became follower at term 0
raft2024/08/27 19:44:11 INFO: newRaft 5 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/08/27 19:44:11 INFO: 5 became follower at term 1
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=(1)
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=(1 2)
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=(1 2 3)
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=(1 2 3 4)
raft2024/08/27 19:44:11 INFO: 5 switched to configuration voters=(1 2 3 4 5)
    node_test.go:91: commits failed to converge!
--- FAIL: TestRestart (5.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16695136?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16695136?buildTab=artifacts#/) on master @ [8551145a0c99c4c95a28ec470e699d0c20ca97ab](https://github.com/cockroachdb/cockroach/commits/8551145a0c99c4c95a28ec470e699d0c20ca97ab):


```
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/08/31 04:10:50 INFO: 2 not forwarding to itself at term 2; dropping proposal
    node_test.go:91: commits failed to converge!
I240831 04:10:55.441900 2411 (gostd) node.go:104  [-] 8  raft.1: stop
I240831 04:10:55.441975 2548 (gostd) node.go:104  [-] 9  raft.2: stop
I240831 04:10:55.441992 2546 (gostd) node.go:104  [-] 10  raft.3: stop
I240831 04:10:55.442054 2417 (gostd) node.go:104  [-] 11  raft.4: stop
I240831 04:10:55.442081 2451 (gostd) node.go:104  [-] 12  raft.5: stop
--- FAIL: TestRestart (5.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16724410?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16724410?buildTab=artifacts#/) on master @ [344fb32842be378bc82c2f3ed8ba12d6f3225be7](https://github.com/cockroachdb/cockroach/commits/344fb32842be378bc82c2f3ed8ba12d6f3225be7):


```
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
raft2024/09/02 19:40:56 INFO: 2 not forwarding to itself at term 3; dropping proposal
    node_test.go:91: commits failed to converge!
I240902 19:41:01.775546 1399 (gostd) node.go:104  [-] 8  raft.1: stop
I240902 19:41:01.775603 1463 (gostd) node.go:104  [-] 9  raft.2: stop
I240902 19:41:01.775618 1461 (gostd) node.go:104  [-] 10  raft.3: stop
I240902 19:41:01.775629 1405 (gostd) node.go:104  [-] 11  raft.4: stop
--- FAIL: TestRestart (5.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

raft/rafttest.TestRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16758400?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16758400?buildTab=artifacts#/) on master @ [77606cbe1c98b248ea658105d7109479439a037f](https://github.com/cockroachdb/cockroach/commits/77606cbe1c98b248ea658105d7109479439a037f):


```
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
raft2024/09/04 22:29:37 INFO: 2 not forwarding to itself at term 2; dropping proposal
    node_test.go:91: commits failed to converge!
--- FAIL: TestRestart (5.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/raft/rafttest/BUILD.bazel
+++ b/pkg/raft/rafttest/BUILD.bazel
@@ -59,6 +59,5 @@ go_test(
     deps = [
         "//pkg/raft",
         "//pkg/raft/raftpb",
-        "//pkg/testutils/skip",
     ],
 )

--- a/pkg/raft/rafttest/node_test.go
+++ b/pkg/raft/rafttest/node_test.go
@@ -24,7 +24,6 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/raft"
 	pb "github.com/cockroachdb/cockroach/pkg/raft/raftpb"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 )
 
 func TestBasicProgress(t *testing.T) {
@@ -54,9 +53,6 @@ func TestBasicProgress(t *testing.T) {
 }
 
 func TestRestart(t *testing.T) {
-	// TODO(pav-kv): de-flake it. See https://github.com/etcd-io/raft/issues/181.
-	skip.UnderStress(t, "the test is flaky")
-
 	peers := []raft.Peer{{ID: 1, Context: nil}, {ID: 2, Context: nil}, {ID: 3, Context: nil}, {ID: 4, Context: nil}, {ID: 5, Context: nil}}
 	nt := newRaftNetwork(1, 2, 3, 4, 5)
 
@@ -136,15 +132,14 @@ func TestPause(t *testing.T) {
 }
 
 func waitLeader(ns []*node) int {
-	var l map[pb.PeerID]struct{}
-	var lindex int
-
+	l := make(map[pb.PeerID]struct{})
 	for {
-		l = make(map[pb.PeerID]struct{})
+		clear(l)
+		lindex := -1
 
 		for i, n := range ns {
 			lead := n.Status().HardState.Lead
-			if lead != 0 {
+			if lead != raft.None {
 				l[lead] = struct{}{}
 				if n.id == lead {
 					lindex = i
@@ -152,7 +147,7 @@ func waitLeader(ns []*node) int {
 			}
 		}
 
-		if len(l) == 1 {
+		if len(l) == 1 && lindex != -1 {
 			return lindex
 		}
 	}

